### PR TITLE
BeeCoin shop & Loadout additions

### PR DESCRIPTION
**Pull request by Agameofscones**
## About The Pull Request

This PR merges the cosmetics & accessories (neck & eyewear) category in the BeeCoin shop, adds a new footwear tab, and fleshes out the options in every area except for the OOC tab while fixing some minor bugs with previous loadout items along the way.

It also disables the associated sort helper proc when building loadout lists, which removes the alphabetical sort, but doesn't remove any functionality from the existing system.

Also changes the descriptions of two armor vests.

Some meme items included, for high prices of course.

Need feedback on things to **ADD** or **REMOVE** from the categories and their lists.

## Why It's Good For The Game

I've personally always found that player expression is an incredibly important part of the roleplay environment. While it is more realistic for everyone to be suited up in uniform and carry on with their day like real people on a real space station, that's a far cry from what SS13 really is and achieves to be with its verisimilitude.

None of the items offered provide any sort of significant bonus, and can already be obtained from different vendor-drobes and lockers around the map (with some exceptions). All alternative mechanical items are entirely role restricted and are more expensive to boot.

The loadout system was ported and consequently abandoned content wise. People have a lot of BeeCoins to spend, and while saving up for a real antag token is great, I think having a few items for expression on a per character basis along the way is a nice thing for people to strive for.

## Changelog
:cl: Triiodine
fix: Hardhats and Station Engineer specific loadout items now actually get added to your loadout  if selected.
add: Added a whole bunch of items to every tab in the BeeCoin loadout shop.
code: Removed the associated sorting of BeeCoin shop category tabs. Aka things aren't alphabetical anymore.
spellcheck: Changed the description of the alternate armor vest, and large armor vest, 
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
remove the Vape, please. it's a very useful item

**Comment by Agameofscones** [comment]
> remove the Vape, please. it's a very useful item

resolved

**Comment by zeskorion** [comment]
> > remove the Vape, please. it's a very useful item
> 
> resolved

Iunno about jackboots and i think winter boots, storage slots and all. Tinfoil hat also seems a bit high cost considering the huge downsides it has, compared to its meager upsides, and lack of sheer memery speedwagon hat has

**Comment by Agameofscones** [comment]
> > > remove the Vape, please. it's a very useful item
> > 
> > 
> > resolved
> 
> Iunno about jackboots and i think winter boots, storage slots and all. Tinfoil hat also seems a bit high cost considering the huge downsides it has, compared to its meager upsides, and lack of sheer memery speedwagon hat has

Winterboots are available in the arrival shuttle lockers, but I'll remove the jackboots and the tinfoil hat.

File: code/modules/client/loadout/loadout.dm [hunk 1]
```diff
@@ -37,9 +37,9 @@ GLOBAL_LIST_EMPTY(gear_datums)
 		LC.gear[use_name] = GLOB.gear_datums[use_name]
 
 	GLOB.loadout_categories = sortAssoc(GLOB.loadout_categories)
-	for(var/loadout_category in GLOB.loadout_categories)
-		var/datum/loadout_category/LC = GLOB.loadout_categories[loadout_category]
-		LC.gear = sortAssoc(LC.gear)
+	//for(var/loadout_category in GLOB.loadout_categories)
```
Comment (ike709) [inline code/modules/client/loadout/loadout.dm]: Don't comment out code, remove it.

File: code/modules/client/loadout/loadout_general.dm [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
 	description = "Says \"Made in China\" on the back. Seems fishy."
 	path = /obj/item/coin/antagtoken
 	cost = 10000
+
+//LIGHTERS & CIGARETTES
+
+/datum/gear/misc/lighter
+	display_name = "Zippo lighter"
+	path = /obj/item/lighter
+	cost = 800
```
Comment (ike709) [inline code/modules/client/loadout/loadout_general.dm]: Should be more expensive IMO.

File: code/modules/client/loadout/loadout_hat.dm [hunk 3]
```diff
@@ -44,4 +185,25 @@
...
+	path = /obj/item/clothing/head/cueball
+	cost = 5000
+
+/datum/gear/hat/piratehat
+	display_name = "pirate hat"
+	description = "Yarr. Comes with one free pirate speak manual."
+	path = /obj/item/clothing/head/pirate
+	cost = 5000
+
+/datum/gear/hat/tinfoil
```
Comment (ike709) [inline code/modules/client/loadout/loadout_hat.dm]: I don't know if this should be an option, but I'm leaning towards no.

**Comment by ike709** [comment]
You need to fix these:
image

See #1701 for details.

**Review by ike709 — CHANGES_REQUESTED** [review]
Do the things.

File: code/modules/client/loadout/loadout_accessories.dm [hunk 4]
```diff
@@ -79,3 +86,65 @@
 	slot = SLOT_GLASSES
 	path = /obj/item/clothing/glasses/monocle
 	cost = 1200
+
+/datum/gear/accessory/glasses
+	display_name = "prescription glasses"
+	slot = SLOT_GLASSES
+	path = /obj/item/clothing/glasses/regular
+	cost = 1200
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_accessories.dm]: Up to 3000

File: code/modules/client/loadout/loadout_general.dm (hunk 2 above)
Comment (qwertyquerty) [reply code/modules/client/loadout/loadout_general.dm]: agree

File: code/modules/client/loadout/loadout_general.dm [hunk 5]
```diff
@@ -1,10 +1,64 @@
...
+
+/datum/gear/misc/lighter/random
+	display_name = "random cheap lighter"
+	path = /obj/item/lighter/greyscale
+	cost = 400
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_general.dm]: more expensiv

File: code/modules/client/loadout/loadout_general.dm [hunk 6]
```diff
@@ -1,10 +1,64 @@
...
+
+/datum/gear/misc/cigarettes
+	display_name = "cigarettes, Space Cigarettes"
+	path = /obj/item/storage/fancy/cigarettes
+	cost = 600
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_general.dm]: more expensiv

File: code/modules/client/loadout/loadout_general.dm [hunk 7]
```diff
@@ -1,10 +1,64 @@
...
+	path = /obj/item/storage/fancy/cigarettes/cigpack_carp
+
+/datum/gear/misc/cigarettes/cigpack_cannabis
+	display_name = "cigarettes, Freak Brothers' Special"
+	path = /obj/item/storage/fancy/cigarettes/cigpack_cannabis
+
+/datum/gear/misc/cigarettes/cohiba
+	display_name = "cigar, Cohiba Robusto"
+	path = /obj/item/clothing/mask/cigarette/cigar/cohiba
+	cost = 1000
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_general.dm]: more expensiv

File: code/modules/client/loadout/loadout_general.dm [hunk 8]
```diff
@@ -1,10 +1,64 @@
...
+
+/datum/gear/misc/cigarettes/havana
+	display_name = "cigar, premium Havanian"
+	path = /obj/item/clothing/mask/cigarette/cigar/havana
+	cost = 1000
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_general.dm]: more expensiv

File: code/modules/client/loadout/loadout_hat.dm [hunk 9]
```diff
@@ -5,32 +5,173 @@
 	species_blacklist = list("plasmaman") //Their helmet takes up the head slot
 	cost = 800
 
+//HARDHATS
+
 /datum/gear/hat/hhat_yellow
 	display_name = "hardhat, yellow"
 	path = /obj/item/clothing/head/hardhat
-	allowed_roles = list("Chief Engineer", "Engineer")
-	cost = 900
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_hat.dm:12]: more expensiv plz

File: code/modules/client/loadout/loadout_hat.dm (hunk 3 above)
Comment (qwertyquerty) [reply code/modules/client/loadout/loadout_hat.dm]: agree

File: code/modules/client/loadout/loadout_suit.dm [hunk 10]
```diff
@@ -3,23 +3,173 @@
 	slot = SLOT_WEAR_SUIT
 	sort_category = "External Wear"
 
+//ALT ARMOR, HAZARD & MEDICAL VESTS, + LABCOATS
+
+/datum/gear/suit/vests/hazard_vest
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_suit.dm]: remove

File: code/modules/client/loadout/loadout_suit.dm [hunk 11]
```diff
@@ -3,23 +3,173 @@
...
+	display_name = "hazard vest"
+	path = /obj/item/clothing/suit/hazardvest
+	allowed_roles = list("Chief Engineer", "Station Engineer", "Atmospheric Technician", "Quartermaster", "Cargo Technician")
+	cost = 2500
+
+/datum/gear/suit/vests/armor/alt //just the alternate appearance armor vest that is in sec lockers.
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_suit.dm]: remove

File: code/modules/client/loadout/loadout_suit.dm [hunk 12]
```diff
@@ -3,23 +3,173 @@
...
+	display_name = "type I-B armor vest"
+	path = /obj/item/clothing/suit/armor/vest/alt
+	allowed_roles = list("Security Officer", "Warden", "Head of Security", "Head of Personnel")
+	cost = 5000
+
+/datum/gear/suit/vests/armor/blueshirt //about that beer I owe ya
```
Comment (qwertyquerty) [inline code/modules/client/loadout/loadout_suit.dm]: remove

**Comment by qwertyquerty** [comment]
We don't want any game changing items. We only want purely cosmetic stuff

File: code/modules/client/loadout/loadout_suit.dm (hunk 12 above)
Comment (ike709) [reply code/modules/client/loadout/loadout_suit.dm]: This is a reskin that can be purchased from the sec vendor, no armor changes.

File: code/modules/client/loadout/loadout_suit.dm (hunk 11 above)
Comment (ike709) [reply code/modules/client/loadout/loadout_suit.dm]: Reskin with no different stats.

**Comment by Agameofscones** [comment]
> Do the things.

Everything addressed.
Fixes everything https://github.com/BeeStation/BeeStation-Hornet/pull/1701 covers with subtypes.

**Pull request merged**